### PR TITLE
[Keyboard] Add "Uno" rev2

### PR DESCRIPTION
**Pull request by spencer-p**
## Description

Adds support for the second revision of Uno, courtesy of @Snipeye.

## Types of Changes

- [x] Keyboard (addition or update)

## Issues Fixed or Closed by This PR

* N/A

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- N/A I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by Snipeye** [comment]
I was going to say something about modifying the default keymap, but then I saw that you put it as a "demo" keymap - really appreciate the work you put in, thank you.

**Review by drashna — APPROVED** [review]

File: keyboards/uno/info.json [hunk 1]
```diff
@@ -1,11 +1,13 @@
 {
     "keyboard_name": "Uno",
+    "keyboard_folder": "uno",
     "url": "https://www.reddit.com/r/mechmarket/comments/gyijm7/gb_uno/",
     "maintainer": "Snipeye",
     "width": 1,
     "height": 1,
```
Comment (fauxpark) [inline keyboards/uno/info.json:7]:
```suggestion
```

File: keyboards/uno/info.json [hunk 2]
```diff
@@ -1,11 +1,13 @@
...
     "layouts": {
       "LAYOUT": {
+        "key_count": 1,
```
Comment (fauxpark) [inline keyboards/uno/info.json:10]:
```suggestion
```

File: keyboards/uno/readme.md [hunk 3]
```diff
@@ -1,19 +1,38 @@
...
+sure to change `rev2` to your Uno's version):
 
- Flashing example for this keyboard:
+    make uno/rev2:default
 
-     make uno:default:flash
+Flashing example for this keyboard (change `rev2` to `rev1` if needed):
 
- See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
+	make uno/rev2:default:flash
```
Comment (fauxpark) [inline keyboards/uno/readme.md:21]:
```suggestion
    make uno/rev2:default:flash
```

File: keyboards/uno/readme.md [hunk 4]
```diff
@@ -1,19 +1,38 @@
...
+
+Create your own keymap with:
+
+	qmk new-keymap -kb uno/rev2
+
+See the [build environment
+setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make
+instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more
+information. Brand new to QMK? Start with our [Complete Newbs
+Guide](https://docs.qmk.fm/#/newbs).
```
Comment (fauxpark) [inline keyboards/uno/readme.md:35]:
```suggestion
See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
```

File: keyboards/uno/readme.md [hunk 5]
```diff
@@ -1,19 +1,38 @@
...
- See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
+	make uno/rev2:default:flash
+
+If you prefer to use the `qmk` tool:
+
+	qmk compile -kb uno/rev2 -km default
+
+Create your own keymap with:
+
+	qmk new-keymap -kb uno/rev2
```
Comment (fauxpark) [inline keyboards/uno/readme.md:29]:
```suggestion
    qmk compile -kb uno/rev2 -km default

Create your own keymap with:

    qmk new-keymap -kb uno/rev2
```

File: keyboards/uno/rev1/config.h [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+#pragma once
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0xFEED
+#define PRODUCT_ID      0xACC8
+#define DEVICE_VER      0x0001
+#define MANUFACTURER    Broekhuijsen
+#define PRODUCT         Uno
+#define DESCRIPTION     A one-key board for the lulz
```
Comment (fauxpark) [inline keyboards/uno/rev1/config.h:26]:
```suggestion
```

File: keyboards/uno/rev1/rules.mk [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+MCU = atmega32u4
+
+# Bootloader selection
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
```
Comment (fauxpark) [inline keyboards/uno/rev1/rules.mk:11]:
```suggestion
```

File: keyboards/uno/rev1/rules.mk [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes        # Enable keyboard RGB underglow
+MIDI_ENABLE = no            # MIDI support
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output
+FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
+HD44780_ENABLE = no         # Enable support for HD44780 based LCDs
```
Comment (fauxpark) [inline keyboards/uno/rev1/rules.mk:32]:
```suggestion
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
```

File: keyboards/uno/rev2/config.h [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+#pragma once
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0xFEED
+#define PRODUCT_ID      0xACC8
+#define DEVICE_VER      0x0001
+#define MANUFACTURER    Broekhuijsen
+#define PRODUCT         Uno
+#define DESCRIPTION     A one-key board for the lulz
```
Comment (fauxpark) [inline keyboards/uno/rev2/config.h:26]:
```suggestion
```

File: keyboards/uno/rev2/rules.mk [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+MCU = atmega32u4
+
+# Bootloader selection
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
```
Comment (fauxpark) [inline keyboards/uno/rev2/rules.mk:11]:
```suggestion
```

File: keyboards/uno/rev2/rules.mk [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes        # Enable keyboard RGB underglow
+MIDI_ENABLE = no            # MIDI support
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output
+FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
+HD44780_ENABLE = no         # Enable support for HD44780 based LCDs
+ENCODER_ENABLE = yes        # Enable rotary encoder support for rev2.
```
Comment (fauxpark) [inline keyboards/uno/rev2/rules.mk:33]:
```suggestion
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
ENCODER_ENABLE = yes
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by spencer-p** [comment]
@fauxpark Merged your changes and fixup'd into the original commit.

File: keyboards/uno/rev2/rules.mk [hunk 12]
```diff
@@ -0,0 +1,23 @@
...
+EXTRAKEY_ENABLE = no       # Audio control and System control
+CONSOLE_ENABLE = no        # Console for debug
+COMMAND_ENABLE = no        # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes        # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
```
Comment (fauxpark) [inline keyboards/uno/rev2/rules.mk:21]:
```suggestion
```

File: keyboards/uno/rev1/rules.mk [hunk 13]
```diff
@@ -0,0 +1,22 @@
...
+EXTRAKEY_ENABLE = no       # Audio control and System control
+CONSOLE_ENABLE = no        # Console for debug
+COMMAND_ENABLE = no        # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes        # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
```
Comment (fauxpark) [inline keyboards/uno/rev1/rules.mk:21]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/uno/uno.h [hunk 14]
```diff
@@ -13,12 +13,13 @@
...
-#define LAYOUT( \
-  K00  \
-) { \
-  { K00 } \
-}
+#ifdef KEYBOARD_uno_rev1
+    #include "rev1.h"
+#elif KEYBOARD_uno_rev2
+    #include "rev2.h"
+#endif
```
Comment (drashna) [inline keyboards/uno/uno.h:25]:
```suggestion
#if defined(KEYBOARD_uno_rev1)
#    include "rev1.h"
#elif defined(KEYBOARD_uno_rev2)
#    include "rev2.h"
#endif
```

**Comment by spencer-p** [comment]
Committed all requested changes; Github still thinks @fauxpark changes are outstanding (??)

**Comment by spencer-p** [comment]
@fauxpark Can you verify your requested changes have been accepted?

File: keyboards/uno/rev2/rules.mk [hunk 15]
```diff
@@ -0,0 +1,22 @@
+# MCU name
+MCU = atmega32u4
+
+# Bootloader selection
+BOOTLOADER = caterina
+
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = no       # Virtual DIP switch configuration
```
Comment (fauxpark) [inline keyboards/uno/rev2/rules.mk:10]:
```suggestion
BOOTMAGIC_ENABLE = no       # Enable Bootmagic Lite
```

File: keyboards/uno/rev1/rules.mk [hunk 16]
```diff
@@ -0,0 +1,21 @@
+# MCU name
+MCU = atmega32u4
+
+# Bootloader selection
+BOOTLOADER = caterina
+
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = no       # Virtual DIP switch configuration
```
Comment (fauxpark) [inline keyboards/uno/rev1/rules.mk:10]:
```suggestion
BOOTMAGIC_ENABLE = no       # Enable Bootmagic Lite
```

File: keyboards/uno/keymaps/demo/keymap.c [hunk 17]
```diff
@@ -0,0 +1,123 @@
...
+                uint16_t timeElapsed = timer_elapsed(pressTimer);
+                switch (presetCounter) {
+                    case 0:
+                        SEND_STRING(SS_LCMD("n"));
+                    break;
+                    case 1:
+                        SEND_STRING("Hello!");
+                    break;
+                    case 2:
+                        SEND_STRING(SS_TAP(X_ENTER)SS_TAP(X_ENTER)"I am uno!");
```
Comment (fauxpark) [inline keyboards/uno/keymaps/demo/keymap.c:63]:
```suggestion
                        SEND_STRING("\n\nI am uno!");
```

File: keyboards/uno/keymaps/demo/keymap.c [hunk 18]
```diff
@@ -0,0 +1,123 @@
...
+                    break;
+                    case 3:
+                        SEND_STRING(SS_TAP(X_ENTER)SS_TAP(X_ENTER)"I can do all sorts of useless things!");
```
Comment (fauxpark) [inline keyboards/uno/keymaps/demo/keymap.c:66]:
```suggestion
                        SEND_STRING("\n\nI can do all sorts of useless things!");
```

File: keyboards/uno/keymaps/demo/keymap.c [hunk 19]
```diff
@@ -0,0 +1,123 @@
...
+                    break;
+                    case 4:
+                        SEND_STRING(SS_TAP(X_ENTER)SS_TAP(X_ENTER)"And I have a built-in RGB LED!"SS_TAP(X_ENTER)SS_TAP(X_ENTER)SS_TAP(X_ENTER));
```
Comment (fauxpark) [inline keyboards/uno/keymaps/demo/keymap.c:69]:
```suggestion
                        SEND_STRING("\n\nAnd I have a built-in RGB LED!\n\n\n");
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by spencer-p** [comment]
@fauxpark Can you confirm there are no outstanding changes required?

File: keyboards/uno/rev2/rules.mk [hunk 20]
```diff
@@ -0,0 +1,22 @@
...
+#
+BOOTMAGIC_ENABLE = no       # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = no       # Mouse keys
+EXTRAKEY_ENABLE = no       # Audio control and System control
+CONSOLE_ENABLE = no        # Console for debug
+COMMAND_ENABLE = no        # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
```
Comment (fauxpark) [inline keyboards/uno/rev2/rules.mk:18]:
```suggestion
NKRO_ENABLE = no            # Enable N-Key Rollover
```

File: keyboards/uno/rev1/rules.mk [hunk 21]
```diff
@@ -0,0 +1,21 @@
...
+#
+BOOTMAGIC_ENABLE = no       # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = no       # Mouse keys
+EXTRAKEY_ENABLE = no       # Audio control and System control
+CONSOLE_ENABLE = no        # Console for debug
+COMMAND_ENABLE = no        # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
```
Comment (fauxpark) [inline keyboards/uno/rev1/rules.mk:18]:
```suggestion
NKRO_ENABLE = no            # Enable N-Key Rollover
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by spencer-p** [comment]
@fauxpark Done

File: keyboards/uno/info.json [hunk 22]
```diff
@@ -1,5 +1,6 @@
 {
     "keyboard_name": "Uno",
+    "keyboard_folder": "uno",
```
Comment (noroadsleft) [inline keyboards/uno/info.json:3]: Use of `keyboard_folder` breaks the API that supports QMK Configurator.

```suggestion
```

File: keyboards/uno/rev1/config.h [hunk 23]
```diff
@@ -0,0 +1,79 @@
...
+
+#pragma once
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0xFEED
+#define PRODUCT_ID      0xACC8
+#define DEVICE_VER      0x0001
+#define MANUFACTURER    Broekhuijsen
+#define PRODUCT         Uno
```
Comment (noroadsleft) [inline keyboards/uno/rev1/config.h:25]: I prefer if the revisions are designated more specifically.

```suggestion
#define PRODUCT         Uno rev1
```

File: keyboards/uno/rev2/config.h [hunk 24]
```diff
@@ -0,0 +1,53 @@
...
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0xFEED
+#define PRODUCT_ID      0xACC8
+#define DEVICE_VER      0x0001
```
Comment (noroadsleft) [inline keyboards/uno/rev2/config.h:23]: Changes the `DEVICE_VER` so it's different between rev1 and rev2.

```suggestion
#define DEVICE_VER      0x0002
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]

File: keyboards/uno/rev2/config.h [hunk 25]
```diff
@@ -0,0 +1,53 @@
...
+#define RGB_DI_PIN D1
+#define RGBLED_NUM 1
+#endif
+
+/* Debounce reduces chatter (unintended double-presses) - set 0 if debouncing is not needed */
+#define DEBOUNCE 5
+
+/* disable these deprecated features by default */
+#define NO_ACTION_MACRO
+#define NO_ACTION_FUNCTION
```
Comment (fauxpark) [inline keyboards/uno/rev2/config.h:53]:
```suggestion
```

File: keyboards/uno/rev1/config.h [hunk 26]
```diff
@@ -0,0 +1,79 @@
...
+//   #define RGBLIGHT_EFFECT_BREATHE_MAX    255   // 0 to 255
+// #endif
+
+/* Debounce reduces chatter (unintended double-presses) - set 0 if debouncing is not needed */
+#define DEBOUNCE 5
+
+/* disable these deprecated features by default */
+#define NO_ACTION_MACRO
+#define NO_ACTION_FUNCTION
+
```
Comment (fauxpark) [inline keyboards/uno/rev1/config.h:79]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by spencer-p** [comment]
@fauxpark @noroadsleft All suggested changes committed.

**Review by noroadsleft — APPROVED** [review]

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**